packages/nx/src/utils/params.ts

```diff
@@ -633,6 +633,8 @@ export async function combineOptionsForGenerator(
     schema,
     false
   );
+
+  warnDeprecations(combined, schema);
```

Comment:
This was being executed after the smart defaults were applied. That was wrong because it would warn about a deprecated option with a smart default (e.g. `project`). The order of operations after this change matches the order of those in the Angular CLI.